src/java/org/apache/cassandra/config/DataStorage.java

```diff
@@ -0,0 +1,378 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.config;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.google.common.primitives.Ints;
+
+/**
+ * Represents an amount of data storage
+ */
+public final class DataStorage
```

Comment:
Thank you for pointing this out!
As far as I can see `parseFileUtils` is implemented for MiBs, KiBs, etc, and we use KB, MB, I am not sure that this change will be wanted. 
About `stringifyFileSize`, it can be used, but it sets a space between unit and quantity and it uses again MiB, KiB etc. Also, our toString will return the quantity with the unit set and not the biggest one.